### ISSUE TITLE
'Unexpected token ILLEGAL' error by a generated JS code

### ISSUE THREAD

**Issue by nebuta** [purescript/purescript#1767]
`/examples/todo` in [https://github.com/zrho/purescript-optic-ui]() compiles, but the generated JS file causes an illegal token error in a browser.

After running `npm run example-todo` in the repo above, I looked into the JS code generated. It has an invalid identifier `wʹ` in `extendFn` as follows.

 `output/Control.Extend/index.js`
```javascript
// Generated by psc version 0.8.0.0
"use strict";
var Prelude = require("Prelude");
var Extend = function (__superclass_Prelude$dotFunctor_0, extend) {
    this["__superclass_Prelude.Functor_0"] = __superclass_Prelude$dotFunctor_0;
    this.extend = extend;
};
var extendFn = function (dictSemigroup) {
    return new Extend(function () {
        return Prelude.functorFn;
    }, function (f) {
        return function (g) {
            return function (w) {
                return f(function (wʹ) {
                    return g(Prelude["<>"](dictSemigroup)(w)(wʹ));
                });
            };
        };
    });
};
var extend = function (dict) {
    return dict.extend;
};
var $less$less$eq = function (dictExtend) {
    return extend(dictExtend);
};
var $eq$less$eq = function (dictExtend) {
    return function (f) {
        return function (g) {
            return function (w) {
                return f($less$less$eq(dictExtend)(g)(w));
            };
        };
    };
};
var $eq$greater$eq = function (dictExtend) {
    return function (f) {
        return function (g) {
            return function (w) {
                return g($less$less$eq(dictExtend)(f)(w));
            };
        };
    };
};
var $eq$greater$greater = function (dictExtend) {
    return function (w) {
        return function (f) {
            return $less$less$eq(dictExtend)(f)(w);
        };
    };
};
var duplicate = function (dictExtend) {
    return extend(dictExtend)(Prelude.id(Prelude.categoryFn));
};
module.exports = {
    Extend: Extend, 
    duplicate: duplicate, 
    "=<=": $eq$less$eq, 
    "=>=": $eq$greater$eq, 
    "=>>": $eq$greater$greater, 
    "<<=": $less$less$eq, 
    extend: extend, 
    extendFn: extendFn
};
```

In contrast, this code generates JS code with an escaped identifier `w$prime`, and works fine (as we expect).
```
module Main where

import Control.Monad.Eff.Console
import Control.Extend

main = do
  log "Hello sailor!"
```

PureScript compiler used is 0.8.0.0. Hope this comment is clear enough to address the issue.

**Comment by garyb** [comment]
@paf31 This is already fixed actually, as of #1737.

**Comment by garyb** [comment]
@nebuta thanks for the report, this will be fixed in the next RC/release.

**Comment by michaelficarra** [comment]
Just curious, what browser didn't like these identifiers?

**Comment by mostalive** [comment]
I'm getting this in Chromium,  "purescript": "0.8.0-rc.1" (which was the latest I could get from npm).

 var iÊ¹ = Prelude["<>"](dictMonoid["__superclass_Prelude.Semigroup_0"]())(i)(Data_Sequence_Internal.measure(dictMeasured)(a));

**Comment by michaelficarra** [comment]
@mostalive You are attempting to use an invalid binding (in JavaScript). Did you create that identifier or did the PureScript compiler generate it for you? Additionally, it looks like there might have been an encoding problem involved. Is this the source code that was given directly by the PureScript compiler, or have you then served it through a web browser or transferred it in some way?

**Comment by mostalive** [comment]
@michaelficarra I eventually managed to get the compiler version I had from stack running on it, and the problem is indeed solved :-). For some strange reason I had two terminal windows referring to the same psc executable, and both gave different output on --version. closing the terminal with the old version fixed it. Never saw anything quite like it. with 0.8.0.0 the problem has vanished. 
FYI This is what I saw in firefox and chrome, so not directly from the purescript compiler. The identifier was generated by the compiler. The Ê¹ seemed to be the character that was givne problems.
